modules/fbs/fbs.module

```diff
@@ -198,6 +198,61 @@ function fbs_settings_form() {
     '#description' => t('Allow users to delete their reservations as well as ready for pickup ones.'),
   );
 
+  // Update list of branches.
+  $form['update'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Update from library system'),
+  );
+
+  $form['update']['fbs_branches_last_updated'] = array(
+    '#markup' => '<label>Last updated:</label>' . (variable_get('fbs_branches_last_updated', 0) ? date('Y-m-d H:i:s', variable_get('fbs_branches_last_updated')) : '<em>' . t('Never') . '</em>'),
+  );
+
+  $form['update']['fbs_update'] = array(
+    '#type' => 'actions',
+    'submit' => array(
+      '#type' => 'submit',
+      '#value' => t('Update list'),
+    ),
+  );
+
+  // Branches and blacklisted branches.
+  $form['branches'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Holdings Branches Blacklist'),
+    '#description' => t('Choose which branches to blacklist from searches'),
+  );
+  $fbs_branches = variable_get('fbs_branches', array());
```

Comment:
No...

Use `fbs_branches()`, and either kill the update button (why do we need it?) or implement caching in `fbs_branches()`.